### PR TITLE
Feature/service observable

### PR DESCRIPTION
**Pull request by akhilarv**

### CODE REVIEW

File: package-lock.json [hunk 1]
```diff
@@ -7496,6 +7526,11 @@
         "tslib": "^1.9.0"
       }
```
Comment (anoopnadesan) [inline package-lock.json:7527]: Delete package lock file

File: src/app/_components/todo/todo-detail/todo-detail.component.html [hunk 2]
```diff
@@ -50,6 +50,15 @@
                             </div>
                         </div>
                     </div>
+                    <div class="form-group row">
+                        <label for="todoImage" class="col-md-2 col-form-label">Uplaod Image</label>
+                        <div class="col-md-5">
+                            <input type="file" class="form-control" #todoImage="ngModel" name="todoImage" [(ngModel)]="todoDetail.todoImage" placeholder="Phone">
+                            <!-- <div [hidden]="phone.valid || (phone.pristine && !todoForm.submitted)" class="text-danger"> -->
```
Comment (anoopnadesan) [inline src/app/_components/todo/todo-detail/todo-detail.component.html:57]: Remove commented code

File: src/app/app.module.ts [hunk 3]
```diff
@@ -12,18 +12,21 @@ import { appRoutingModule } from './app.routing';
 import { JwtInterceptor, ErrorInterceptor } from './_helpers';
 import { HomeComponent } from './_components/home';
 import { LoginComponent } from './_components/login';
+import { SharedModules } from './shared_modules/shared.modules';
```
Comment (anoopnadesan) [inline src/app/app.module.ts:15]: Rename folder shared_modules to shared

File: src/app/app.routing.ts [hunk 4]
```diff
@@ -1,4 +1,5 @@
 ﻿import { Routes, RouterModule } from '@angular/router';
+import { MatTableModule } from '@angular/material';
```
Comment (anoopnadesan) [inline src/app/app.routing.ts:2]: Double check if it's required.

File: src/app/shared_modules/footer/footer.component.html [hunk 5]
```diff
@@ -0,0 +1,7 @@
+
+<!-- Footer -->
+<div class="text-center mt-4">
+  <p>
+      <a href="{{homeUrl}}" target="_top">wwwapps.biz</a>
```
Comment (anoopnadesan) [inline src/app/shared_modules/footer/footer.component.html:5]: Removed hardcoding of wwwapps.biz text

File: src/app/shared_modules/header/header.component.ts [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+  constructor(private authenticationService: AuthenticationService, private router: Router) {
+    this.authenticationService.currentUser.subscribe(x => this.currentUser = x);
+  }
+
+  logout() {
+    this.authenticationService.logout();
+    this.router.navigate(['/login']);
+  }
+  ngOnInit() {
+
```
Comment (anoopnadesan) [inline src/app/shared_modules/header/header.component.ts:23]: empty ngOnInit function can be removed.

File: src/app/shared_modules/footer/footer.component.ts [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+  selector: 'app-footer',
+  templateUrl: './footer.component.html',
+  styleUrls: ['./footer.component.less']
+})
+export class FooterComponent implements OnInit {
+  
+  homeUrl = 'http://wwwapps.biz/';
+  constructor() { }
+
+  ngOnInit() {
```
Comment (anoopnadesan) [inline src/app/shared_modules/footer/footer.component.ts:13]: remove pls

Comment (akhilarv) [reply src/app/shared_modules/footer/footer.component.ts:13]: fixed

File: src/app/shared_modules/header/header.component.ts (hunk 6 above)
Comment (akhilarv) [reply src/app/shared_modules/header/header.component.ts:23]: fixed

File: src/app/shared_modules/footer/footer.component.html (hunk 5 above)
Comment (akhilarv) [reply src/app/shared_modules/footer/footer.component.html:5]: fixed

File: src/app/app.routing.ts (hunk 4 above)
Comment (akhilarv) [reply src/app/app.routing.ts:2]: fixed

File: src/app/app.module.ts (hunk 3 above)
Comment (akhilarv) [reply src/app/app.module.ts:15]: fixed

File: src/app/_components/todo/todo-detail/todo-detail.component.html (hunk 2 above)
Comment (akhilarv) [reply src/app/_components/todo/todo-detail/todo-detail.component.html:57]: fixed

File: package-lock.json (hunk 1 above)
Comment (akhilarv) [reply package-lock.json:7527]: fixed

**Pull request merged**